misc-utils/rename.c

```diff
@@ -112,11 +131,17 @@ static int do_file(char *from, char *to, char *s, int verbose, int noact, int no
 		file = s;
 	if (string_replace(from, to, file, s, &newname))
 		return 0;
-	if (nooverwrite && access(newname, F_OK) == 0) {
+
+	if ((nooverwrite || interactive) && access(newname, F_OK) != 0)
+		nooverwrite = interactive = 0;
+	if (nooverwrite) {
 		printf(_("Skipping existing file: `%s'\n"), newname);
 		ret = 0;
 	}
-	else if (!noact && rename(s, newname) != 0) {
+	else if ( !noact &&
+	          ( (interactive && ask(newname) != 0) ||
+	            rename(s, newname) != 0 ) )
+	{
 		warn(_("%s: rename to %s failed"), s, newname);
 		ret = 2;
```

Comment:
It prints "rename failed" error message and return 2 when ask() returns zero. I don't think it's right thing. It would be better to use something like:
```
else if (noact)
  ; /* to nothing */
else if (interactive && ask(newname) != 0)
  printf(_("Skipping %s\n", newname);
else if (rename(s, newname) != 0) {
  warn(_("%s: rename to %s failed"), s, newname);
  ret = 2;
}
```